src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -154,22 +205,9 @@ private bool WriteStream(TKey key, Stream stream, Checksum? checksum, Cancellati
                 return false;
             }
 
-            private Stream? ReadBlob(
-                SqlConnection connection, Database database,
-                TDatabaseId dataId, string columnName,
-                Checksum? checksum, CancellationToken cancellationToken)
+            private Optional<Stream> ReadDataBlob(
+                SqlConnection connection, Database database, long rowId, Checksum? checksum)
             {
-                // Note: it's possible that someone may write to this row between when we
-                // get the row ID above and now.  That's fine.  We'll just read the new
-                // bytes that have been written to this location.  Note that only the
-                // data for a row in our system can change, the ID will always stay the
-                // same, and the data will always be valid for our ID.  So there is no
-                // safety issue here.
-                if (!TryGetRowId(connection, database, dataId, out var rowId))
-                {
-                    return null;
-                }
```

Comment:
this logic moved into the caller.  it now computes this and passes in the rowid to us.